client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -18,31 +18,70 @@ import { updateTeam } from '../../actions/team';
 const ProfileInfoEdit = () => {
   const dispatch = useDispatch();
 
-  const { teamId, teamName, orgName, email } = useSelector(
+  const { teamId, teamName, orgName, email,password, confirmedPassword } = useSelector(
     (state) => state.team
   );
 
-  const [profileData, setInputs] = useState({ teamName, orgName, email });
+  const [profileData, setInputs] = useState({ teamName, orgName, email,password ,confirmedPassword});
 
   useEffect(() => {
-    setInputs({ teamName, orgName, email });
+    setInputs({ teamName, orgName, email,password,confirmedPassword  });
   }, [email, orgName, teamName]);
 
   const updateInputs = (form) => {
     const { name, value } = form.target;
-    setInputs({ ...profileData, [name]: value });
+    if (name !== "email" ){
+      setInputs({ ...profileData, [name]: value });
+    }
   };
+  const checkPassword = function(){
+    console.log({ ...profileData}.password);
 
-  const [validated, setValidated] = useState(false);
-  const handleUpdate = (event) => {
-    const form = event.currentTarget;
-    event.preventDefault();
-    if (form.checkValidity() === false) {
-      event.stopPropagation();
-    } else {
-      dispatch(updateTeam(teamId, profileData));
+    if (! ({...profileData}.password == {...profileData}.confirmedPassword)){
+      return false;
     }
-    setValidated(true);
+    var matches = {...profileData}.password.match(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/);
+    if (matches == null){
+      console.log("false");
+      return false;
+    }
+    return true;
+
+  };
+
+  const [validated, setValidated] = useState(false);
+  const handleUpdate = async (event) => {
+      const form = event.currentTarget;
+      event.preventDefault();
+      if (form.checkValidity() === false) {
+          event.stopPropagation();
+      } else {
+          if (({...profileData}.password || {...profileData}.confirmedPassword ) && ({...profileData}.password != "" || {...profileData}.confirmedPassword != "")) {
+              if (!checkPassword()) {
+                    setValidated(false);
+                    alert("please enter a password with numbers, letters and characters, at least 8 chars long")
+                    return;
+              } else {
+                  var newdata = {
+                      "teamName": {...profileData}.teamName,
+                      "orgName": {...profileData}.orgName,
+                      "email": {...profileData}.email,
+                      "password": {...profileData}.password
+                  }
+                  dispatch(updateTeam(teamId, newdata));
+                  alert("a");
```

Comment:
Please add a meaningful message here instead of `a`. Could be something like `Password successfully updated.`